src/Workspaces/CoreTest/UtilityTest/EditDistanceTests.cs

```diff
@@ -7,87 +7,256 @@ namespace Microsoft.CodeAnalysis.UnitTests
+                "thousand","three","through","throw","thus","tie","time","tiny","tire","to","together","told","tone",
+                "too","took","tool","top","total","touch","toward","town","track","trade","train","travel","tree","triangle",
+                "trip","trouble","truck","true","try","tube","turn","twenty","two","type","under","unit","until","up",
+                "us","use","usual","valley","value","vary","verb","very","view","village","visit","voice","vowel","wait",
+                "walk","wall","want","war","warm","was","wash","watch","water","wave","way","we","wear","weather","week",
+                "weight","well","went","were","west","what","wheel","when","where","whether","which","while","white",
+                "who","whole","whose","why","wide","wife","wild","will","win","wind","window","wing","winter","wire",
+                "wish","with","woman","women","wonder","wont","wood","word","work","world","would","write","written",
+                "wrong","wrote","yard","year","yellow","yes","yet","you","young","your",
+            };
+
+
         [Fact]
-        public void TestNotCloseMatch()
+        public void Top1000Test()
         {
-            Assert.False(EditDistance.IsCloseMatch("propertyblocksyntax", "ipropertysymbol"));
-            Assert.False(EditDistance.IsCloseMatch("propertyblocksyntax", "ipropertysymbolextensions"));
-            Assert.False(EditDistance.IsCloseMatch("propertyblocksyntax", "typeblocksyntaxextensions"));
+            for (var i = 0; i < Top1000.Length; i++)
+            {
+                var source = Top1000[i];
+                for (var j = 0; j < Top1000.Length; j++)
+                {
+                    var target = Top1000[j];
+                    var editDistance1 = EditDistance.GetEditDistance(source, target);
 
-            Assert.False(EditDistance.IsCloseMatch("fielddeclarationsyntax", "declarationinfo"));
-            Assert.False(EditDistance.IsCloseMatch("fielddeclarationsyntax", "declarationcomputer"));
-            Assert.False(EditDistance.IsCloseMatch("fielddeclarationsyntax", "filelinepositionspan"));
+                    if (i == j)
+                    {
+                        Assert.Equal(0, editDistance1);
+                    }
 
-            Assert.False(EditDistance.IsCloseMatch("variabledeclaratorsyntax", "visualbasicdeclarationcomputer"));
-            Assert.False(EditDistance.IsCloseMatch("variabledeclaratorsyntax", "ilineseparatorservice"));
+                    Assert.True(editDistance1 >= 0);
 
-            Assert.False(EditDistance.IsCloseMatch("expressionsyntax", "awaitexpressioninfo"));
+                    var editDistance2 = EditDistance.GetEditDistance(source, target, editDistance1);
+                    Assert.Equal(editDistance1, editDistance2);
+                }
+            }
+        }
+
+        [Fact]
+        public void TestMetric()
```

Comment:
The previous test is testing 3 of 4 aspects of a metric space (symmetry, non-negativity, and identity of indiscernibles). This test is checking only the 4th (triangle inequality), and this one is the most likely to fail in a non-metric implementation of edit distance. Can you expand on this test to make it try all triples from some set? (You check the top 1000 words for symmetry, which is 1,000,000 pairs. You could use the top 100 terms to check 1,000,000 triples.)